gax/src/main/java/com/google/api/gax/tracing/OpencensusTracer.java

```diff
@@ -342,6 +345,10 @@ public void batchRequestSent(long elementCount, long requestSize) {
           "attempt response count", AttributeValue.longAttributeValue(attemptReceivedMessages));
     }
 
+    if (lastConnectionId != null) {
```

Comment:
Should the referrence be copied first (for atomicity), as it is technically not guaranteed to be same on liness 348 and 349?

Reply:
done